workspaces/ui-v2/src/optic-components/pages/diffs/AddEndpointsPage/AddEndpointsPage.tsx

```diff
@@ -158,6 +160,54 @@ export function DiffUrlsPage(props: any) {
   );
 }
 
+const PendingEndpointItem: FC<{
+  endpoint: IPendingEndpoint;
+}> = ({ endpoint }) => {
+  const diffReviewPagePendingEndpoint = useDiffReviewPagePendingEndpoint();
+  const changelogStyles = useChangelogStyles();
+  const classes = useStyles();
+  const history = useHistory();
+  const { forceRerender } = useSharedDiffContext();
+
+  const [state]: any = useActor(endpoint.ref);
```

Comment:
To subscribe to changes here, we need to use the `useActor` hook. However in the `Save Changes` button, we need to group together all the states of all pending endpoints. Due to how hooks work, we can't actually do a `for endpoint, useActor loop`

This code below effectively _recreates_ what happens in react if the endpoint ref was _not_ transparent and did cause rerenders.

It's also possible there is an easier way to do this (the only "real" way I could think through here was to remodel the state + pull up the isLoading state into the sharedDiffStore and access it there).

Reply:
I think you are correct - however it's not deterministic / ordered since this would loop over pendingEndpoints - which is an array that changes as we add more pendingEndpoints.